src/AspNet.Security.OAuth.Nextcloud/NextcloudAuthenticationHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+
+
+namespace AspNet.Security.OAuth.Nextcloud
+{
+    public class NextcloudAuthenticationHandler : OAuthHandler<NextcloudAuthenticationOptions>
+    {
+        public NextcloudAuthenticationHandler(
+            [NotNull] IOptionsMonitor<NextcloudAuthenticationOptions> options,
+            [NotNull] ILoggerFactory logger,
+            [NotNull] UrlEncoder encoder,
+            [NotNull] ISystemClock clock)
+            : base(options, logger, encoder, clock)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
+            string usr = tokens.Response.SelectToken("user_id").ToString();
+            string usrentpoint = Options.UserInformationEndpoint.TrimEnd('/');
+            usrentpoint += $"/{usr}";
```

Comment:
Does this need to be URI-encoded?

Reply:
yes because you need to directly call the user.
```
USERID = username
GET /ocs/v1.php/cloud/users/USERID
```

Reply:
So in that case you need to do `Uri.EscapeDataString(...)` (or `Uri.EscapeUriString(...)`) before you format the value into the URL.